torchaudio/csrc/sox_utils.cpp

```diff
@@ -0,0 +1,71 @@
+#include <c10/core/ScalarType.h>
+#include <sox.h>
+#include <torchaudio/csrc/sox_utils.h>
+
+namespace torchaudio {
+namespace sox_utils {
+
+caffe2::TypeMeta get_dtype(
+    const sox_encoding_t encoding,
+    const unsigned precision) {
+  const auto dtype = [&]() {
+    switch (encoding) {
+      case SOX_ENCODING_UNSIGNED: // 8-bit PCM WAVE
+        return torch::kUInt8;
+      case SOX_ENCODING_SIGN2: // 16-bit or 32-bit PCM WAVE
+        switch (precision) {
+          case 16:
+            return torch::kInt16;
+          case 32:
+            return torch::kInt32;
+          default:
+            throw std::runtime_error(
+                "Only 16 and 32 bits are supported for signed PCM.");
+        }
+      case SOX_ENCODING_FLOAT: // 32-bit floating-point WAVE
+      case SOX_ENCODING_MP3:
+      case SOX_ENCODING_FLAC:
+      case SOX_ENCODING_VORBIS:
+      case SOX_ENCODING_OPUS:
```

Comment:
This limits the codec supported, right? Are we reducing the support from what was mentioned in the readme? should we make an update to the [readme](https://github.com/pytorch/audio/blob/master/README.md) too?

Reply:
This would be BC-breaking then, right?

Reply:
This is not yet BC-breaking since this PR doesn't replace the current backend, but creates a new one instead.

Reply:
I gave some thoughts on this one, and I am okay with relaxing this, given that we clearly state somewhere (probably README AND document) that `wav`, `mp3`, `flac` and `vorbis` are the only formats rigorously tested and that the other formats might work but the correctness is not tested.

The original reason I wanted to limit the formats is that I wanted to make it clear that only some of the formats are verified its correctness, so that users are aware of it.

I think just passing through the other formats and the behavior should be same as the existing "sox" backend.